examples/c-comparisons/script.jl

```diff
@@ -0,0 +1,125 @@
+# # Binary Classification with Laplace approximation
+#
+# This example demonstrates how to carry out non-conjugate Gaussian process
+# inference using the Laplace approximation.
+#
+# For a basic introduction to the functionality of this library, please refer
+# to the [User Guide](@ref).
+#
+# ## Setup
+
+using ApproximateGPs
+using LinearAlgebra
+using Distributions
+using LogExpFunctions: logistic, softplus, invsoftplus
+#using ParameterHandling
+using Zygote
+using Optim
+
+using Plots
+default(; legend=:outertopright, size=(700, 400))
+
+using Random
+Random.seed!(1);
+
+# ## Generate some training data
+#
+# We create a binary-labelled toy dataset:
+
+Xgrid = -4:0.1:29  # for visualization
+X = range(0, 23.5; length=48)  # training inputs
+f(x) = 3 * sin(10 + 0.6x) + sin(0.1x) - 1  # latent function
+fs = f.(X)  # latent function values at training inputs
+const invlink = logistic  # could use other invlink, e.g. normcdf(f) = cdf(Normal(), f)
+ps = invlink.(fs)  # probabilities at the training inputs
+Y = [rand(Bernoulli(p)) for p in ps]  # observations at the training inputs
+
+function plot_data()
+    plot(; xlims=extrema(Xgrid), xticks=0:6:24)
+    plot!(Xgrid, invlink ∘ f; label="true probabilities")
+    return scatter!(X, Y; label="observations", color=3)
+end
+
+plot_data()
+
+# ## Creating the latent GP
+# 
+# Here we write a function that creates our latent GP prior, given the
+# hyperparameter vector `theta`. Compared to a "vanilla" GP, the `LatentGP`
+# requires a function or functor that maps from the latent GP `f` to the
+# distribution of observations `y`. This functor is commonly called "the
+# likelihood".
+
+function build_latent_gp(theta)
+    # `theta` is unconstrained, but kernel variance and lengthscale must be positive:
+    variance = softplus(theta[1])
+    lengthscale = softplus(theta[2])
+
+    kernel = variance * with_lengthscale(SqExponentialKernel(), lengthscale)
+
+    dist_y_given_f = BernoulliLikelihood()  # has logistic invlink by default
+    # We could also be explicit and define it as a function:
+    # dist_y_given_f(f) = Bernoulli(invlink(f))
+
+    jitter = 1e-8  # required for numeric stability [TODO: where to explain this better?]
+    return LatentGP(GP(kernel), dist_y_given_f, jitter)
+end
+
+# We define a latent GP at our initial hyperparameter values, here with
+# variance 1.0 and lengthscale 5.0:
+
+theta0 = [invsoftplus(1.0), invsoftplus(5.0)]
+
+lf = build_latent_gp(theta0)
+
+lf.f.kernel
+
+# We can now compute the posterior `p(f | y)` under the Laplace approximation:
+
+f_post = posterior(LaplaceApproximation(), lf(X), Y)
```

Comment:
Maybe leave a comment here to point out that calling this runs an optimisation problem? Maybe just helpful for the reader to know where / when stuff happens.

Reply:
done, happy?